hack/test/e2e-capi.sh

```diff
@@ -4,10 +4,13 @@ set -eou pipefail
 
 source ./hack/test/e2e.sh
 
-export CAPI_VERSION="0.3.19"
+# See https://github.com/talos-systems/talos/issues/4046
+export K8S_VERSION=1.21.3
```

Comment:
I _think_ that is wrong place. Basically 1.21.x should be installed in the `e2e-docker` step for the CAPI pipeline. 

Workload cluster can perfectly be 1.22.x, but management cluster should be 1.21.x

In our CI pipeline, management cluster - `e2e-docker` (`talosctl cluster create` Docker cluster), workload cluster - GCP or AWS